### PR TITLE
ci: update pyright to 1.1.300 in CI

### LINKED ISSUE #589: CI: update CI to most recent Pyright versions
As discussed in #588 the update to v1.1.300 seems to have broken our stuff. The changelog can be found here: https://github.com/microsoft/pyright/releases/tag/1.1.300

When pyrigth v 1.1.300 is happy again, one can jsut revert the above PR to let the CI use the most uptodate version again.

### PR DESCRIPTION
**Pull request by superlopuh**
Fixes #589

### CODE REVIEW

File: xdsl/dialects/cmath.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ class Mul(Operation):
 
     # TODO replace with trait
     def verify_(self) -> None:
-        if self.lhs != self.rhs.typ and self.rhs.typ != self.result.typ:
+        if self.lhs.typ != self.rhs.typ and self.rhs.typ != self.result.typ:
```
Comment (superlopuh) [inline xdsl/dialects/cmath.py:52]: 👻

File: .github/workflows/ci-pyright-fails.yml [hunk 2]
```diff
@@ -34,4 +34,4 @@ jobs:
       - name: Pyright
         uses: jakebailey/pyright-action@v1
         with:
-          version: 1.1.299
+          version: 1.1.300
```
Comment (superlopuh) [inline .github/workflows/ci-pyright-fails.yml:37]: I propose to keep the version fixed so we have control over updating the version

Comment (webmiche) [reply .github/workflows/ci-pyright-fails.yml:37]: Could we actually integrate this with something like dependabot thing? Something that opens a PR on a new version, but does not directly merge it. Maybe if we stopped using the pyright action, made pyright an optional requirement, and then call it from the cmd line.

**Review by webmiche — APPROVED** [review]

File: .github/workflows/ci-pyright-fails.yml (hunk 2 above)
Comment (superlopuh) [reply .github/workflows/ci-pyright-fails.yml:37]: Sounds like another issue to me :)

**Review by math-fehr — APPROVED** [review]
Nice! Way less changes than we expected ;)

File: .github/workflows/ci-pyright-fails.yml (hunk 2 above)
Comment (webmiche) [reply .github/workflows/ci-pyright-fails.yml:37]: #630

**Pull request merged**